modules/hostap/src/supp_api.c

```diff
@@ -2586,3 +2595,235 @@ int supplicant_config_params(const struct device *dev, struct wifi_config_params
 	k_mutex_unlock(&wpa_supplicant_mutex);
 	return ret;
 }
+
+#ifdef CONFIG_WIFI_NM_WPA_SUPPLICANT_P2P
+static inline void extract_value(const char *src, char *dest, size_t dest_size)
+{
+	size_t i = 0;
+
+	while (src[i] != '\0' && src[i] != '\n' && i < dest_size - 1) {
+		dest[i] = src[i];
+		i++;
+	}
+	dest[i] = '\0';
+}
+
+static void parse_peer_info_line(const char *line, struct wifi_p2p_device_info *info)
+{
+	const char *pos;
+
+	if (strncmp(line, "device_name=", 12) == 0) {
+		extract_value(line + 12, info->device_name, sizeof(info->device_name));
+	} else if (strncmp(line, "pri_dev_type=", 13) == 0) {
+		extract_value(line + 13, info->pri_dev_type_str, sizeof(info->pri_dev_type_str));
+	} else if (strncmp(line, "level=", 6) == 0) {
+		pos = line + 6;
+		/* Skip extraction for numeric values, atoi handles it */
+		info->rssi = (int8_t)atoi(pos);
```

Comment:
We are mixing `atoi()` and `strtol()` and the atoi one cannot detect errors in the input string. I suggest we use only `strtol()` and make sure that the string is a valid one (now we ignore the errors).